### PR TITLE
Fix layout spacing problems

### PR DESCRIPTION
**Pull request by w568w**
Cherry picked from #303 for 1.4.0.

### CODE REVIEW

**Comment by singularity-s0** [comment]
Exact what problem does this PR fix?

**Comment by w568w** [comment]
> Exact what problem does this PR fix?

Too big/small paddings, layout overflows... Something like these.

File: lib/page/subpage_danke.dart [hunk 1]
```diff
@@ -75,94 +79,105 @@ class DankeSubPageState extends PlatformSubpageState<DankeSubPage> {
...
+          key: indicatorKey,
+          edgeOffset: MediaQuery.of(context).padding.top,
+          color: Theme.of(context).colorScheme.secondary,
+          backgroundColor: Theme.of(context).dialogBackgroundColor,
+          onRefresh: () async {
+            HapticFeedback.mediumImpact();
+            await _loadRandomReview(forceRefetch: true);
+            setState(() {});
+          },
+          child: SizedByChildBuilder(
```
Comment (w568w) [inline lib/page/subpage_danke.dart:92]: Why use `SizedByChildBuilder` here, as you do not seem to use the measured `size` later at all?

cc @HydrogenC

File: lib/page/subpage_danke.dart [hunk 2]
```diff
@@ -75,94 +79,105 @@ class DankeSubPageState extends PlatformSubpageState<DankeSubPage> {
...
-                    stackTrace: snapshot.stackTrace,
-                    onTap: () => setState(() {}));
-              }
-            },
-            loadingBuilder: Center(
-              child: PlatformCircularProgressIndicator(),
-            ))
-        // A maximized result list
-        : Expanded(child: CourseListWidget(searchKeyword: searchText));
+    return Expanded(
```
Comment (w568w) [inline lib/page/subpage_danke.dart:139]: Again, do not use `Expanded` as your root widget in a method or a custom widget's `build()`. 

cc @HydrogenC

**Review by w568w — COMMENTED** [review]
Minor changes.

File: lib/page/subpage_danke.dart (hunk 1 above)
Comment (HydrogenC) [reply lib/page/subpage_danke.dart]: The top padding used to be height-dependent of search bar, but I forgot to remove this after the top padding was cancelled.

File: lib/page/subpage_danke.dart (hunk 2 above)
Comment (HydrogenC) [reply lib/page/subpage_danke.dart]: My bad.

**Pull request merged**